src/components/Data/Documents/Page.vue

```diff
@@ -390,6 +405,41 @@ export default {
     }
   },
   methods: {
+    async subscribeToCurrentDocs() {
+      try {
+        if (this.subscribeRoomId) {
+          await this.$kuzzle.realtime.unsubscribe(this.subscribeRoomId)
+          this.subscribeRoomId = null
+        }
+        const roomId = await this.$kuzzle.realtime.subscribe(
+          this.indexName,
+          this.collectionName,
+          {
+            or: this.documents.map(d => ({ equals: { _id: d._id } }))
+          },
+          this.realtimeNotifCallback
+        )
+        this.subscribeRoomId = roomId
+      } catch (error) {
+        this.$log.error(error)
+      }
+    },
+    realtimeNotifCallback(notif) {
+      if (notif.scope === 'out') {
+        return
+      }
+      if (!['replace', 'update'].includes(notif.action)) {
+        return
+      }
+      const documents = JSON.parse(JSON.stringify(this.documents))
+      const idx = documents.findIndex(d => d._id === notif.result._id)
+      documents[idx] = {
+        ...documents[idx],
+        ...notif.result._source
```

Comment:
Why don't you just mutate the concerned document? You might need a `this.$set`

Reply:
there is actually a $set but on the whole documents array because since it's `deep` the event wasn't spread to children props
We can discuss about that on discord next we if you want